library/src/main/java/io/ona/kujaku/manager/DrawingManager.java

```diff
@@ -106,27 +107,29 @@ public void onAnnotationDragFinished(Circle circle) {
             }
         });
 
-        mapboxMap.addOnMapClickListener(new MapboxMap.OnMapClickListener() {
-            @Override
-            public boolean onMapClick(@NonNull LatLng point) {
-                final PointF pixel = mapboxMap.getProjection().toScreenLocation(point);
-                List<Feature> features = mapboxMap.queryRenderedFeatures(pixel, (Expression) null, CircleManager.ID_GEOJSON_LAYER);
-
-                if (features.size() == 0 && drawingEnabled) {
-                    if (getCurrentKujakuCircle() != null) {
-                        unsetCurrentCircleDraggable();
-                    } else {
-                        drawCircle(point);
+        if(drawOnClick) {
```

Comment:
Why did you add this boolean param? It doesn't seem to serve any purpose according to the library module code. Are you by any chance using it any other way?

Reply:
Could you also format the file, there should be space between `if` and `(`, codacy should have caught this format/style violation

Reply:
> Why did you add this boolean param? It doesn't seem to serve any purpose according to the library module code. Are you by any chance using it any other way?

Yes, I am adding these circles as user is walking a certain path without user having to click the map. 

> Could you also format the file, there should be space between if and (, codacy should have caught this format/style violation

Sure, My default style settings are bit different, so avoiding to format the whole file.

Reply:
@ekigamba removed this constructor param since I realized I can get the same behaviour by not calling `startDrawing` and instead direcly calling `drawCircle` as user moves the boundary

Reply:
Awesome